src/reporting/Market.sol

```diff
@@ -209,7 +213,20 @@ contract Market is DelegationTarget, Typed, Initializable, Ownable, IMarket {
         // FIXME: if finalPayoutDistributionHash == getIdentityDistributionId(), transfer validity bond to reportingWindow (reporter fee pot)
         // FIXME: if automated report is wrong, transfer automated report bond to reportingWindow
         // FIXME: if automated report is right, transfer automated report bond to market creator
-        // FIXME: handle markets that get 0 reports during their scheduled reporting window. We should move to the next reporting window in this case.
+    }
+
+    // This function should not use the ReportingState since we may be in AWAITING_FINALIZATION which would make the method of determining reporting ambiguous 
+    function hasReceivedAReport() private constant returns (bool) {
+        if (getBranch().getForkingMarket() == this) {
+            return true;
+        }
+        if (address(limitedReportersDisputeBondToken) != NULL_ADDRESS) {
+            return reportingWindow.hasReceivedAllReport(this);
+        }
+        if (automatedReportReceivedTime == 0 || address(automatedReporterDisputeBondToken) != NULL_ADDRESS) {
+            return reportingWindow.getNumberOfReportsByMarket(this) > 0;
+        }
+        return true;
```

Comment:
Currently, this will return true if the market hasn't ended yet or is still in designated reporting phase.  I recommend ensuring this method returns the correct value in _all_ situations, or throws in situations it can't handle, or the method name is changed to make it very clear when it is appropriate to call this method.  I worry that even if its current usage pattern is such that it is only called under fairly tight constraints, a future update may relax those and the code will _appear_ correct when it no longer is.

Reply:
If we add another state as suggested below I think we can actually remove this function entirely.